lib/apiml.js

```diff
@@ -202,6 +203,93 @@ ApimlConnector.prototype = {
     });
   },
   
+  getInstanceId() {
+    return this.zluxServerEurekaClient.config.instance.instanceId;
+  },
+  
+  getZluxInstances() {
+    const eurekaClient = this.zluxServerEurekaClient;
+    return new Promise((resolve, reject) => {
+      eurekaClient.once('registryUpdated', () => {
+        const zluxInstances = eurekaClient.getInstancesByVipAddress('ZLUX');
+        log.debug(`registry updated, zluxInstances ${JSON.stringify(zluxInstances, null, 2)}`);
+        zluxInstances.sort((a, b) => a.instanceId.localeCompare(b.instanceId));
+        resolve(zluxInstances);
+      });
+    });
+  },
+  
+  waitUntilZluxClusterIsReady(clusterSize) {
+    return this.getZluxInstances().then(
+      zluxInstances => {
+        if (zluxInstances.length < clusterSize) {
+          return this.waitUntilZluxClusterIsReady(clusterSize);
+        }
+        return zluxInstances;
+      }
+    )
+  },
+
+  takeIntoService() {
+    log.debug(`takeIntoService`);
+    return this._deleteOverriddenStatus();
+  },
+  
+  takeInstanceIntoService(instanceId) {
+    log.info(`takeInstanceIntoService ${instanceId}`);
+    return this._deleteOverriddenStatusForInstance(instanceId);
+  },
+  
+  takeOutOfService() {
+    log.debug(`takeOutOfservice`);
+    return this._overrideStatus('OUT_OF_SERVICE');
+  },
+  
+  takeInstanceOutOfService(instanceId) {
+    log.debug(`takeInstanceOutOfService ${instanceId}`);
+    return this._overrideStatusForInstance(instanceId, 'OUT_OF_SERVICE');
+  },
+  
+  _overrideStatus(status) {
+    return this._overrideStatusForInstance(this.zluxServerEurekaClient.instanceId, status);
+  },
+  
+  _overrideStatusForInstance(instanceId, status) {
+    return new Promise((resolve, reject) => {
+      this.zluxServerEurekaClient.eurekaRequest({
+        method: 'PUT',
+        uri: `${this.zluxServerEurekaClient.config.instance.app}/${instanceId}/status?value=${status}`
```

Comment:
is OUT_OF_SERVICE a special name to eureka/apiml, or is it just something that was chosen for the app-server

Reply:
Eureka server recognizes a few statuses — `UP`, `DOWN`, `STARTING`, `OUT_OF_SERVICE` and `UNKNOWN`.

Reply:
It's best to define those as consts somewhere so that people will know what the choices are for future code.